engine/classes/Elgg/Database/Mutex.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Elgg\Database;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * Provides database mutex that can be used to prevent race conditions
+ * between two processes that affect the same data.
+ *
+ * @access private
+ * @since  2.1.0
+ */
+class Mutex {
+
+	/**
+	 * @var \Elgg\Database
+	 */
+	private $db;
+
+	/**
+	 * @var \Elgg\Logger
+	 */
+	private $logger;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \Elgg\Database $db
+	 * @param \Elgg\Logger   $logger
+	 */
+	public function __construct(\Elgg\Database $db, \Elgg\Logger $logger) {
+		$this->db = $db;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * Creates a table {prefix}{$namespace}_lock that is used as a mutex.
+	 *
+	 * @param string $namespace Allows having separate locks for separate processes
+	 * @return bool
+	 */
+	public function lock($namespace) {
+		if (!$this->isLocked($namespace)) {
```

Comment:
Even though this is private API, how about validating $namespace with ctype_alpha()?